src/signals/settings/users/containers/Detail/__tests__/Detail.test.js

```diff
@@ -1,39 +1,192 @@
-      ({ container, rerender } = await render(withAppContext(<UserDetail location={{}} />)));
+      ({ container, rerender } = await render(withAppContext(<UserDetail />)));
     });
 
-    expect(container.querySelector('a').getAttribute('href')).toEqual(routes.users);
+    expect(container.querySelector('a').getAttribute('href')).toEqual(
+      routes.users
+    );
+
+    jest.spyOn(reactRouterDom, 'useLocation').mockImplementationOnce(() => ({
+      referrer,
+    }));
 
     await act(async () => {
-      rerender(withAppContext(<UserDetail location={{ referrer }} />));
+      rerender(withAppContext(<UserDetail />));
     });
 
     expect(container.querySelector('a').getAttribute('href')).toEqual(referrer);
   });
 
-  it('should show an alert on error', async () => {
-    let getByText;
-    const error = new Error();
-    fetch.mockRejectOnce(error);
+  it('should instantiate useFetchUser', () => {
+    render(withAppContext(<UserDetail />));
 
-    await act(async () => {
-      ({ getByText } = await render(withAppContext(<UserDetail location={{}} />)));
-    });
+    expect(useFetchUser).toHaveBeenCalled();
+  });
+
+  it('should render a loading indicator', () => {
+    useFetchUser.mockImplementationOnce(() => ({ isLoading: true }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    expect(getByTestId('loadingIndicator')).toBeInTheDocument();
+  });
+
+  it('should render a form', () => {
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    expect(getByTestId('detailUserForm')).toBeInTheDocument();
+  });
+
+  it('should show an alert on error', () => {
+    const message = 'Something went wrong here';
+    useFetchUser.mockImplementationOnce(() => ({ error: { message } }));
+
+    const { getByTestId, getByText } = render(withAppContext(<UserDetail />));
+
+    expect(getByTestId('formAlert')).toBeInTheDocument();
+    expect(getByText(message)).toBeInTheDocument();
+  });
+
+  it('should not patch user data on submit when form data has not been altered', () => {
+    const patch = jest.fn();
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON, patch }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    fireEvent.click(
+      getByTestId('detailUserForm').querySelector('[type="submit"]')
+    );
+
+    expect(patch).not.toHaveBeenCalled();
+  });
+
+  it('should patch user data on submit', () => {
+    const patch = jest.fn();
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON, patch }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    fireEvent.change(
+      getByTestId('detailUserForm').querySelector('#last_name'),
+      { target: { value: 'Foo Bar Baz' } }
+    );
+    fireEvent.click(
+      getByTestId('detailUserForm').querySelector('[type="submit"]')
+    );
+
+    expect(patch).toHaveBeenCalled();
```

Comment:
Please also test that this has `not.ToHaveBeenCalled()` before the `fireEvent.click()`.